packages/lodestar-validator/src/api/interface/validators.ts

```diff
@@ -1,22 +1,22 @@
 import {
+  AggregateAndProof,
   Attestation,
+  AttestationData,
+  AttesterDuty,
   BeaconBlock,
   BLSPubkey,
   BLSSignature,
   CommitteeIndex,
   Epoch,
-  Slot,
-  ValidatorDuty,
-  SignedBeaconBlock
+  SignedBeaconBlock,
+  Slot
 } from "@chainsafe/lodestar-types";
 
 export interface IValidatorApi {
   
-  getProposerDuties(epoch: Epoch): Promise<Map<Slot, BLSPubkey>>;
+  getProposerDuties(epoch: Epoch, validatorPubKeys: BLSPubkey[]): Promise<Map<Slot, BLSPubkey>>;
 
-  getAttesterDuties(epoch: Epoch, validatorPubKey: BLSPubkey[]): Promise<ValidatorDuty[]>;
-
-  isAggregator(slot: Slot, committeeIndex: CommitteeIndex, slotSignature: BLSSignature): Promise<boolean>;
+  getAttesterDuties(epoch: Epoch, validatorPubKeys: BLSPubkey[]): Promise<AttesterDuty[]>;
 
   /**
    * Requests a BeaconNode to produce a valid block,
```

Comment:
You might rebase off 0.11.x branch, there's a change to `produceBlock` there, which currently adds a BLSPubkey to identify the proposer. Definitely suitable for validatorIndex, too.

Reply:
yeah I need it for #746 as well, please rebase to 0.11.x if possible

Reply:
Sorry I don't get how this api is related to #746 or 0.11.x? in 0.11.x produceBlock method will just add proposer index to BeaconBlock model, it doesn't require validator input on this as it knows index because it knows proposer on given slot?

Reply:
in #746, I need to create `AggregateAndProof` and `SignedAggregateAndProof` from the validator side. In order to do that I need to issue a REST call to get aggregator index which is available in this PR.